src/components/DayPicker/DayPicker.tsx

```diff
@@ -257,33 +258,36 @@ export const DayPicker: React.FC<DayPickerProps> = ({
           [classes.textFieldNoManualInput]: !parseDate,
         })}
         onChange={_onTextChange}
-        onMouseDown={(e) => {
-          /**
-           * This handler allows for opening _and_ closing the popover via click in
-           * readonly mode.
-           *
-           * We need to use onMouseDown instead of onClick so we can run this _before_
-           * onFocus runs and avoid an immediate open/close.
-           */
-          if (!parseDate) {
-            setIsCalendarOpen((current) => !current);
-          }
-
-          textFieldProps.onMouseDown?.(e);
-        }}
-        onFocus={(e) => {
-          setIsCalendarOpen(true);
-          textFieldProps.onFocus?.(e);
-        }}
-        onBlur={(e) => {
-          /**
-           * This logic primarily serves to provide good keyboard navigation.
-           */
-          if (!isInternalNode(e.relatedTarget)) {
-            setIsCalendarOpen(false);
-          }
-          textFieldProps.onBlur?.(e);
-        }}
+        onMouseDown={composeEventHandlers([
+          textFieldProps.onMouseDown,
+          () => {
+            /**
+             * This handler allows for opening _and_ closing the popover via click in
+             * readonly mode.
+             *
+             * We need to use onMouseDown instead of onClick so we can run this _before_
+             * onFocus runs and avoid an immediate open/close.
+             */
+            if (!parseDate) {
+              setIsCalendarOpen((current) => !current);
+            }
+          },
+        ])}
+        onFocus={composeEventHandlers([
+          textFieldProps.onFocus,
+          () => setIsCalendarOpen(true),
+        ])}
+        onBlur={composeEventHandlers([
+          textFieldProps.onBlur,
+          (e) => {
+            /**
+             * This logic primarily serves to provide good keyboard navigation.
+             */
+            if (!isInternalNode(e.relatedTarget)) {
+              setIsCalendarOpen(false);
+            }
+          },
+        ])}
```

Comment:
None of these handlers are really changed, they just use the `composeEventHandlers` helper now.